src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -1211,25 +1211,25 @@ internal static string GetAssemblyFileVersion(Assembly assembly)
 
         protected override void Dispose(bool disposing)
         {
-            base.Dispose(disposing);
-
             if (disposing)
             {
+                (TraceWriter as IDisposable)?.Dispose();
+#if FEATURE_NODE
+                NodeFunctionInvoker.UnhandledException -= OnUnhandledException;
+#endif
                 _scriptFileWatcher?.Dispose();
                 _debugModeFileWatcher?.Dispose();
+                _blobLeaseManager?.Dispose();
 
                 foreach (var function in Functions)
                 {
                     (function.Invoker as IDisposable)?.Dispose();
                 }
-
-                _blobLeaseManager?.Dispose();
-                (TraceWriter as IDisposable)?.Dispose();
-
-#if FEATURE_NODE
-                NodeFunctionInvoker.UnhandledException -= OnUnhandledException;
-#endif
             }
+
+            // dispose base last to ensure that errors there don't
+            // cause us to not dispose ourselves
+            base.Dispose(disposing);
```

Comment:
Issue was that this guy was throwing, preventing us from running our dispose code, leaving the trace writer and other instances live, resulting in the linearly increasing duplicate logs.